doc/mgmt/SONiC_overlay_ConfigDB.md

```diff
@@ -0,0 +1,234 @@
+| ------------------------------------------------------------- | ------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Get default value from Yang model in read API.                | Redis config DB keeps no change.                  | 3 MB memory per-process because need load Yang model and reference libyang.<br>50ms to load yang model.<br>8ms to read default value 10000 times. |
+| Write default value to default value DB when write config DB. | Better read performance, Less memory consumption. | Need add new Redis DB for default value.                                                                                                          |
+
+### How to get static config
+
+- Static config will stored in a new redis database 'STATIC_CONFIG_DB'
+- Static config DB tables will have exactly same name and schema with config DB tables.
+- Data will read form static config DB with swsscommon API.
+- Static config DB will save and persist for warm-reboot and fast-reboot.
+
+### API compatibility
+
+|                                                       | Pros                                                                              | Cons                                                                                                                                     |
+| ----------------------------------------------------- | --------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
+| Change API to return default value and static config. | Less code change, all app will get default value and static config automatically. | For default value, there are hardcoded default value may different with Yang model, new default value from config DB may cause code bug. |
+| Existed API keeps no change.                          | When update existed code, can cleanup code to remove hard coded default value.    | All apps need code update.                                                                                                               |
+
+### Current design:
+
+- Existed read API keeps no change. 
+- Add decorator API to return default value and static config data.
+
+## 2.2 New class
+
+- YangModelLoader class
+  
+  - load table name to default value mapping to memory.
+
+- DefaultValueProvider class
+  
+  - Find default value information by table name and config DB key
+  - Merge default value to API result.
+
+- StaticConfigProvider class
+  
+  - Read static config from static config DB.
+  - Merge static config to API result.
+
+- OverlayConfigDBConnectorDecorator python class
+
+- OverlayConfigTable c++ class
+
+- OverlaySubscriberStateTable c++ class
```

Comment:
>OverlaySubscriberStateTable [](http://example.com/codeflow?start=2&length=27)

To make names consistent, OverlayConfigSubscriberStateTable, or OverlayConfigTable->OverlayTable. Need to think about overlay vs underlay

Reply:
Fixed, rename to UnderlayTable